rust_src/src/strings.rs

```diff
@@ -202,6 +202,32 @@ fn string_to_unibyte(string: LispObject) -> LispObject {
     }
 }
 
+// @TODO need to rework this function to use new as_symbol_or_error API
+fn get_string_or_symbol(mut string: LispObject) -> multibyte::LispStringRef {
+    if string.is_symbol() {
+        string = string.symbol_name()
+    }
+
+    string.as_string_or_error()
+}
+
+#[lisp_fn]
+fn string_lessp(string1: LispObject, string2: LispObject) -> LispObject {
+    let lispstr1 = get_string_or_symbol(string1);
+    let lispstr2 = get_string_or_symbol(string2);
+
+    let mut count = 0;
+    let zip = lispstr1.iter().zip(lispstr2.iter());
+    for ((codept1, _), (codept2, _)) in zip {
+        count += 1;
+        if codept1 != codept2 {
+            return LispObject::from_bool(codept1 < codept2);
+        }
+    }
+
+    LispObject::from_bool(count < lispstr2.len_chars())
```

Comment:
I think this is just `lispstr1.len_chars() < lispstr2.len_chars()`, and keeping `count` is unnecessary.

(If lispstr2 is shorter or equally long, this will test `len2 < len2` which is false.)